src/jit/lsra.cpp

```diff
@@ -4922,14 +5009,21 @@ LinearScan::allocateBusyReg(Interval *current, RefPosition *refPosition)
             // to avoid a spill - codegen can then insert the copy.
             assert(candidates == candidateBit);
             physRegNextLocation = MaxLocation;
+            farthestRefPosWeight = BB_MAX_WEIGHT;
         }
         else
         {
             physRegNextLocation = physRegRecord->getNextRefLocation();
+
+            // If refPosition requires a fixed register, we should reject all others.
+            // Otherwise, we will still evaluate all phyRegs though their next location is
+            // not better than farthestLocation found so far.
+            if (refPosition->isFixedRegRef && physRegNextLocation < farthestLocation)
+            {
+                continue;
+            }
         }
```

Comment:
Please note that the existing code already has a similar check to bail out early. See lines 4930-4931 below.

After my changes, this condition was needed only for fixedRef case because, a reg record needs to be evaluated based on weight instead of farthest distance.  

Sure, I have added a TODO comment here.